### PR TITLE
Update CHANGES.txt for version 3.10.0

### PR DESCRIPTION
**Pull request by rafi-kamal**

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+    
+    First step in migrating getResultBase64String away from deprecated boolean option.
+    
+    (a) this aligns with other base64 API's
+    (b) this removes the need for some clients that strip padding from web safe
+    (c) allows new API for JsProto to use the enum rather than limited boolean options.
+  * Delete jspb.BinaryIterator.
+  * Fix sint64 zig-zag encoding.
+  * Simplify hash64 string conversion to avoid DIGIT array. Should reduce overhead if these functions aren't used, and be more efficient by avoiding linear array searches.
+  * Use jasmine expect instead of closure test asserts.
```
Comment (TeBoring) [inline CHANGES.txt]: not needed

File: CHANGES.txt [hunk 2]
```diff
@@ -1,3 +1,77 @@
...
+  * Change the parameter types of binaryReaderFn in ExtensionFieldBinaryInfo to (number, ?, ?).
+  * Create dates.ts and time_of_days.ts to mirror Java versions. This is a near-identical conversion of c.g.type.util.{Dates,TimeOfDays} respectively.
+  * Migrate moneys to TypeScript.
```
Comment (TeBoring) [inline CHANGES.txt]: Not needed.

File: CHANGES.txt [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+  * Java: Add Automatic-Module-Name entries to the Manifest (#6568)
+
+  Python
+  * Add descriptor methods in descriptor_pool are deprecated.
+  * Uses explicit imports to prevent multithread test failures in py3.
+  * Added __delitem__ for Python extension dict
+  * Update six version to 1.12.0 and fix legacy_create_init issue (#6391)
+
+  JavaScript
+  * Remove deprecated boolean option to getResultBase64String().
```
Comment (TeBoring) [inline CHANGES.txt]: Could you merge all three entries about getResultBase64String? Some seems internal only info

File: CHANGES.txt [hunk 4]
```diff
@@ -1,3 +1,77 @@
+2019-09-03 version 3.10.0 (C++/Java/Python/PHP/Objective-C/C#/Ruby/JavaScript)
+
+  C++
+  * Switch the proto parser to the faster MOMI parser.
+  * Properly escape Struct keys in the proto3 JSON serializer.
+  * Inlined implementation of ..._IsValid routines for enums with small numbers of distinct ranges of values
```
Comment (acozzette) [inline CHANGES.txt]: This enum _IsValid optimization ended up getting rolled back and so it is not included in this release.

File: CHANGES.txt (hunk 1 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done

File: CHANGES.txt (hunk 3 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done

File: CHANGES.txt (hunk 2 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done

File: CHANGES.txt (hunk 1 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Actually, do you mean all four of these are not needed or just "Delete jspb.BinaryInterator" is not needed?

File: CHANGES.txt (hunk 4 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done

**Comment by rafi-kamal** [comment]
I updated the wrong branch and now Github can't detect the changes, although it shows up in https://github.com/protocolbuffers/protobuf/compare/3.10.x...rafi-kamal:3.10.x-201909031846?expand=1.

**Pull request closed without merging**